client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -104,6 +104,11 @@
  */
 public class RestClient implements Closeable {
 
+    /** The version of the Rest client*/
+    // Must be kept in sync with org.elasticsearch.Version.CURRENT, but we can't call that class as the low level
+    // client should not depend on the server code. A test in `VersionTests` will fail if they're not equal.
+    public static final String VERSION = "7.11.0";
```

Comment:
Please do not hardcode this version. Even with a test, it would add another place that needs to be updated on every version bump, increasing the complexity of the bump.  Instead, we can use the manifest attribute for Elasticsearch version that every jar contains. And we should fail if this attribute does not exist, to ensure if a user builds a fat jar with the client, we will not silently be missing the version.

Reply:
This is some telemetry information that users should not be bothered with. They can opt out if they want, but it should not bring additional requirements on their build system. So we'd rather fail silently and report a missing version, hoping that it will not be statistically significant.

And if it is significant (which is why reporting missing information is important too) then we can work on updating the build system so that this information exists as generated code that will survive a fat jar operation.

This is targeted at 7.x is because FF for 7.11.0 is really close and I wanted to avoid the additional time needed for a backport. So 7.x first, and backport to master even if I know this isn't the common way of doing it.